hardhat/env-contract.js

```diff
@@ -1,24 +1,40 @@
-extendEnvironment(env => {
-  const { contract } = env;
+// Remove the default account from the accounts list used in tests, in order
+// to protect tests against accidentally passing due to the contract
+// deployer being used subsequently as function caller
+//
+// This operation affects:
+// - the accounts (and signersAsPromise) parameters of `contract` blocks
+// - the return of hre.ethers.getSigners()
+extendEnvironment(hre => {
+  // TODO: replace with a mocha root hook.
+  // (see https://github.com/sc-forks/solidity-coverage/issues/819#issuecomment-1762963679)
```

Comment:
Honestly I don't see how to use these hooks because they require using a `--require` flag that I don't see exposed through Hardhat's test CLI (based on mocha).